### PR TITLE
terminology-update

### PR DESCRIPTION
**Pull request by LakinS**
Contains all 3 branches of changes, all of which passed X86 tests, haven't checked others yet, on their own, but start failing some tests after merging.

### CODE REVIEW

File: src/cpu/BaseCPU.py [hunk 1]
```diff
@@ -211,17 +211,17 @@ def connectAllPorts(self, cached_bus, uncached_bus = None):
     def addPrivateSplitL1Caches(self, ic, dc, iwc = None, dwc = None):
         self.icache = ic
         self.dcache = dc
-        self.icache_port = ic.cpu_side
-        self.dcache_port = dc.cpu_side
-        self._cached_ports = ['icache.mem_side', 'dcache.mem_side']
+        self.icache_port = ic.cpu_side_port
```
Comment (powerjg) [inline src/cpu/BaseCPU.py]: This should just be cpu_side

File: src/arch/arm/table_walker.cc [hunk 2]
```diff
@@ -55,7 +55,7 @@ using namespace ArmISA;
 
 TableWalker::TableWalker(const Params *p)
     : ClockedObject(p),
-      stage2Mmu(NULL), port(NULL), masterId(Request::invldMasterId),
+      stage2Mmu(NULL), port(NULL), _id(Request::invldUniqueId),
```
Comment (powerjg) [inline src/arch/arm/table_walker.cc:58]: invldRequestorId?

File: src/arch/arm/table_walker.cc [hunk 3]
```diff
@@ -97,11 +97,11 @@ TableWalker::~TableWalker()
 }
 
 void
-TableWalker::setMMU(Stage2MMU *m, MasterID master_id)
+TableWalker::setMMU(Stage2MMU *m, RequestorID unique_id)
```
Comment (powerjg) [inline src/arch/arm/table_walker.cc:100]: requestor_id?

File: src/arch/arm/table_walker.hh [hunk 4]
```diff
@@ -829,8 +829,8 @@ class TableWalker : public ClockedObject
     /** Port shared by the two table walkers. */
     DmaPort* port;
 
-    /** Master id assigned by the MMU. */
-    MasterID masterId;
+    /** Unique id assigned by the MMU. */
+    RequestorID _id;
```
Comment (powerjg) [inline src/arch/arm/table_walker.hh:833]: _requestorId?

File: src/arch/arm/tlb.cc [hunk 5]
```diff
@@ -107,10 +107,10 @@ TLB::init()
 }
 
 void
-TLB::setMMU(Stage2MMU *m, MasterID master_id)
+TLB::setMMU(Stage2MMU *m, RequestorID unique_id)
 {
     stage2Mmu = m;
-    tableWalker->setMMU(m, master_id);
+    tableWalker->setMMU(m, unique_id);
```
Comment (powerjg) [inline src/arch/arm/tlb.cc:113]: requestor_id?

File: src/arch/arm/tlb.hh [hunk 6]
```diff
@@ -224,7 +224,7 @@ class TLB : public BaseTLB
 
     TableWalker *getTableWalker() { return tableWalker; }
 
-    void setMMU(Stage2MMU *m, MasterID master_id);
+    void setMMU(Stage2MMU *m, RequestorID unique_id);
```
Comment (powerjg) [inline src/arch/arm/tlb.hh:227]: requestor_id?

File: src/arch/arm/tracers/tarmac_parser.cc [hunk 7]
```diff
@@ -1287,7 +1287,7 @@ TarmacParserRecord::readMemNoEffect(Addr addr, uint8_t *data, unsigned size,
     ArmISA::TLB* dtb = static_cast<TLB*>(thread->getDTBPtr());
 
     req->setVirt(addr, size, flags, thread->pcState().instAddr(),
-                 Request::funcMasterId);
+                 Request::funcUniqueId);
```
Comment (powerjg) [inline src/arch/arm/tracers/tarmac_parser.cc:1290]: funcRequestorId

File: src/arch/x86/pagetable_walker.cc [hunk 8]
```diff
@@ -519,7 +519,7 @@ Walker::WalkerState::stepWalk(PacketPtr &write)
         Request::Flags flags = oldRead->req->getFlags();
         flags.set(Request::UNCACHEABLE, uncacheable);
         RequestPtr request = std::make_shared<Request>(
-            nextRead, oldRead->getSize(), flags, walker->masterId);
+            nextRead, oldRead->getSize(), flags, walker->_id);
```
Comment (powerjg) [inline src/arch/x86/pagetable_walker.cc:522]: _requestorId?

File: src/arch/x86/pagetable_walker.hh [hunk 9]
```diff
@@ -168,7 +168,7 @@ namespace X86ISA
         // The TLB we're supposed to load.
         TLB * tlb;
         System * sys;
-        MasterID masterId;
+        RequestorID _id;
```
Comment (powerjg) [inline src/arch/x86/pagetable_walker.hh:171]: _requestorId?

File: src/arch/x86/pagetable_walker.hh [hunk 10]
```diff
@@ -204,7 +204,7 @@ namespace X86ISA
         Walker(const Params *params) :
             ClockedObject(params), port(name() + ".port", this),
             funcState(this, NULL, NULL, true), tlb(NULL), sys(params->system),
-            masterId(sys->getMasterId(this)),
+            _id(sys->getUniqueId(this)),
```
Comment (powerjg) [inline src/arch/x86/pagetable_walker.hh:207]: getRequestorId?

File: src/cpu/BaseCPU.py [hunk 11]
```diff
@@ -211,17 +211,17 @@ def connectAllPorts(self, cached_bus, uncached_bus = None):
...
+        self.dcache_port = dc.cpu_side_port
```
Comment (powerjg) [inline src/cpu/BaseCPU.py]: ditto, cpu_side

File: src/cpu/BaseCPU.py [hunk 12]
```diff
@@ -211,17 +211,17 @@ def connectAllPorts(self, cached_bus, uncached_bus = None):
...
+        self._cached_ports = ['icache.mem_side_port', 'dcache.mem_side_port']
```
Comment (powerjg) [inline src/cpu/BaseCPU.py]: should be mem_side for both

File: src/cpu/BaseCPU.py [hunk 13]
```diff
@@ -211,17 +211,17 @@ def connectAllPorts(self, cached_bus, uncached_bus = None):
...
         if buildEnv['TARGET_ISA'] in ['x86', 'arm', 'riscv']:
             if iwc and dwc:
                 self.itb_walker_cache = iwc
                 self.dtb_walker_cache = dwc
-                self.itb.walker.port = iwc.cpu_side
-                self.dtb.walker.port = dwc.cpu_side
-                self._cached_ports += ["itb_walker_cache.mem_side", \
-                                       "dtb_walker_cache.mem_side"]
+                self.itb.walker.port = iwc.cpu_side_port
```
Comment (powerjg) [inline src/cpu/BaseCPU.py]: cpu_side

File: src/cpu/BaseCPU.py [hunk 14]
```diff
@@ -211,17 +211,17 @@ def connectAllPorts(self, cached_bus, uncached_bus = None):
...
+                self.dtb.walker.port = dwc.cpu_side_port
```
Comment (powerjg) [inline src/cpu/BaseCPU.py]: cpu_side

File: src/cpu/BaseCPU.py [hunk 15]
```diff
@@ -211,17 +211,17 @@ def connectAllPorts(self, cached_bus, uncached_bus = None):
...
+                self._cached_ports += ["itb_walker_cache.mem_side_port", \
```
Comment (powerjg) [inline src/cpu/BaseCPU.py]: Both of these should be mem_side

File: src/cpu/BaseCPU.py [hunk 16]
```diff
@@ -237,8 +237,8 @@ def addTwoLevelCacheHierarchy(self, ic, dc, l2c, iwc=None, dwc=None,
         self.toL2Bus = xbar if xbar else L2XBar()
         self.connectCachedPorts(self.toL2Bus)
         self.l2cache = l2c
-        self.toL2Bus.master = self.l2cache.cpu_side
-        self._cached_ports = ['l2cache.mem_side']
+        self.toL2Bus.mem_side_port = self.l2cache.cpu_side_port
```
Comment (powerjg) [inline src/cpu/BaseCPU.py]: mem_side_port is correct, but it should be cpu_side

File: src/cpu/base.cc [hunk 17]
```diff
@@ -122,8 +122,8 @@ CPUProgressEvent::description() const
 
 BaseCPU::BaseCPU(Params *p, bool is_checker)
     : ClockedObject(p), instCnt(0), _cpuId(p->cpu_id), _socketId(p->socket_id),
-      _instMasterId(p->system->getMasterId(this, "inst")),
-      _dataMasterId(p->system->getMasterId(this, "data")),
+      _instUniqueId(p->system->getUniqueId(this, "inst")),
```
Comment (powerjg) [inline src/cpu/base.cc:125]: RequestorId?

File: src/cpu/base.cc [hunk 18]
```diff
@@ -250,7 +250,7 @@ BaseCPU::mwaitAtomic(ThreadID tid, ThreadContext *tc, BaseTLB *dtb)
     if (secondAddr > addr)
         size = secondAddr - addr;
 
-    req->setVirt(addr, size, 0x0, dataMasterId(), tc->instAddr());
+    req->setVirt(addr, size, 0x0, dataUniqueId(), tc->instAddr());
```
Comment (powerjg) [inline src/cpu/base.cc:253]: dataRequestorId?

File: src/cpu/base.hh [hunk 19]
```diff
@@ -124,10 +124,10 @@ class BaseCPU : public ClockedObject
     const uint32_t _socketId;
 
     /** instruction side request id that must be placed in all requests */
-    MasterID _instMasterId;
+    RequestorID _instUniqueId;
```
Comment (powerjg) [inline src/cpu/base.hh:127]: requestorId?

File: src/arch/arm/stage2_mmu.cc [hunk 20]
```diff
@@ -49,13 +49,13 @@ using namespace ArmISA;
 Stage2MMU::Stage2MMU(const Params *p)
     : SimObject(p), _stage1Tlb(p->tlb), _stage2Tlb(p->stage2_tlb),
       port(_stage1Tlb->getTableWalker(), p->sys),
-      masterId(p->sys->getMasterId(_stage1Tlb->getTableWalker()))
+      _id(p->sys->getUniqueId(_stage1Tlb->getTableWalker()))
```
Comment (powerjg) [inline src/arch/arm/stage2_mmu.cc:52]: getRequestorId

File: src/arch/arm/stage2_mmu.hh [hunk 21]
```diff
@@ -60,7 +60,7 @@ class Stage2MMU : public SimObject
     DmaPort port;
 
     /** Request id for requests generated by this MMU */
-    MasterID masterId;
+    RequestorID _id;
```
Comment (powerjg) [inline src/arch/arm/stage2_mmu.hh:63]: I think Andreas wanted this to be _requestorId

File: src/arch/riscv/pagetable_walker.cc [hunk 22]
```diff
@@ -396,7 +396,7 @@ Walker::WalkerState::stepWalk(PacketPtr &write)
     else {
         //If we didn't return, we're setting up another read.
         RequestPtr request = std::make_shared<Request>(
-            nextRead, oldRead->getSize(), flags, walker->masterId);
+            nextRead, oldRead->getSize(), flags, walker->_id);
```
Comment (powerjg) [inline src/arch/riscv/pagetable_walker.cc:399]: _requestorId?

File: src/cpu/base.hh [hunk 23]
```diff
@@ -182,9 +182,9 @@ class BaseCPU : public ClockedObject
     uint32_t socketId() const { return _socketId; }
 
     /** Reads this CPU's unique data requestor ID */
-    MasterID dataMasterId() const { return _dataMasterId; }
+    RequestorID dataUniqueId() const { return _dataUniqueId; }
```
Comment (powerjg) [inline src/cpu/base.hh:185]: requestorId

File: src/cpu/checker/cpu.hh [hunk 24]
```diff
@@ -89,7 +89,7 @@ class CheckerCPU : public BaseCPU, public ExecContext
     using VecRegContainer = TheISA::VecRegContainer;
 
     /** id attached to all issued requests */
-    MasterID masterId;
+    RequestorID _id;
```
Comment (powerjg) [inline src/cpu/checker/cpu.hh:92]: _requestorId?

File: src/cpu/testers/traffic_gen/base.hh [hunk 25]
```diff
@@ -53,7 +53,7 @@ class System;
 struct BaseTrafficGenParams;
 
 /**
- * The traffic generator is a master module that generates stimuli for
+ * The traffic generator is a memside module that generates stimuli for
```
Comment (powerjg) [inline src/cpu/testers/traffic_gen/base.hh:56]: Just "module"

File: src/cpu/testers/traffic_gen/base_gen.cc [hunk 26]
```diff
@@ -46,8 +46,8 @@
 #include "debug/TrafficGen.hh"
 #include "sim/system.hh"
 
-BaseGen::BaseGen(SimObject &obj, MasterID master_id, Tick _duration)
-    : _name(obj.name()), masterID(master_id),
+BaseGen::BaseGen(SimObject &obj, RequestorID unique_id, Tick _duration)
```
Comment (powerjg) [inline src/cpu/testers/traffic_gen/base_gen.cc:49]: requestor_id

File: src/cpu/testers/traffic_gen/base_gen.cc [hunk 27]
```diff
@@ -69,19 +69,19 @@ BaseGen::getPacket(Addr addr, unsigned size, const MemCmd& cmd,
...
-        std::fill_n(pkt_data, req->getSize(), (uint8_t)masterID);
+        std::fill_n(pkt_data, req->getSize(), (uint8_t)_id);
     }
 
     return pkt;
 }
 
 StochasticGen::StochasticGen(SimObject &obj,
-                             MasterID master_id, Tick _duration,
+                             RequestorID unique_id, Tick _duration,
```
Comment (powerjg) [inline src/cpu/testers/traffic_gen/base_gen.cc:79]: requestor_id

File: src/cpu/testers/traffic_gen/base_gen.hh [hunk 28]
```diff
@@ -85,10 +85,10 @@ class BaseGen
      * Create a base generator.
      *
      * @param obj simobject owning the generator
-     * @param master_id MasterID set on each request
+     * @param unique_id RequestorID set on each request
```
Comment (powerjg) [inline src/cpu/testers/traffic_gen/base_gen.hh:88]: requestor_id

File: src/cpu/testers/traffic_gen/base_gen.hh [hunk 29]
```diff
@@ -85,10 +85,10 @@ class BaseGen
...
      * @param _duration duration of this state before transitioning
      */
-    BaseGen(SimObject &obj, MasterID master_id, Tick _duration);
+    BaseGen(SimObject &obj, RequestorID unique_id, Tick _duration);
```
Comment (powerjg) [inline src/cpu/testers/traffic_gen/base_gen.hh:91]: requestor_id

File: src/cpu/testers/traffic_gen/traffic_gen.hh [hunk 30]
```diff
@@ -45,15 +45,15 @@
 struct TrafficGenParams;
 
 /**
- * The traffic generator is a master module that generates stimuli for
+ * The traffic generator is a memside module that generates stimuli for
```
Comment (powerjg) [inline src/cpu/testers/traffic_gen/traffic_gen.hh:48]: just "module"

File: src/mem/cache/base.hh [hunk 31]
```diff
@@ -572,7 +572,7 @@ class BaseCache : public ClockedObject
      * @param pkt The request to perform.
      * @param fromCpuSide from the CPU side port or the memory side port
      */
-    virtual void functionalAccess(PacketPtr pkt, bool from_cpu_side);
```
Comment (LakinS) [inline src/mem/cache/base.hh:575]: Should this be switched back?

Comment (powerjg) [reply src/mem/cache/base.hh:575]: Yeah, I'd switch it back

**Pull request closed without merging**